### ISSUE TITLE
Error raises when use multiprocessing

### ISSUE THREAD

**Issue by takumi-ricoh** [spyder-ide/spyder#8128]
### What steps will reproduce the problem?

This is Error

AttributeError: module '__main__' has no attribute '__spec__'

this code raise the error 
---------------------------------------
from multiprocessing import Pool

a long code　・・・ 

if __name__ == '__main__':

    datalist = [・・・]

    p = Pool(4)
    
    res=p.map(get_data,datalist)
---------------------------------------

On Anaconda Prompt the code runs fine.

this is the same as the following

https://stackoverflow.com/questions/45720153/python-multiprocessing-error-attributeerror-module-main-has-no-attribute

## Versions

* Spyder version: 3.3.1 
* Python version: 3.6.6
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
pyflakes >=0.6.0  :  2.0.0 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
sphinx >=0.6.6    :  1.7.6 (OK)
rope >=0.9.4      :  0.11.0 (OK)
jedi >=0.9.0      :  0.12.1 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  0.23.4 (OK)
numpy >=1.7       :  1.15.0 (OK)
sympy >=0.7.3     :  1.2 (OK)
cython >=0.21     :  0.28.5 (OK)
qtconsole >=4.2.0 :  4.4.0 (OK)
IPython >=4.0     :  6.5.0 (OK)
matplotlib >=2.0.0:  2.2.3 (OK)
pylint >=0.25     :  2.1.1 (OK)
```

**Comment by jnsebgosselin** [comment]
@ccordoba12 this is the bug I was referring to the other day and for which I was planning to eventually fill an issue.

This occurs if you run the code after `Remove all variables` in the IPython console or if you `Execute in a dedicated console`.

The code run fine if you `Execute in current console` and do not `Remove all variables` or if you `Execute in an external system terminal`.

Here is the MWE from the [SO post](https://stackoverflow.com/questions/45720153/python-multiprocessing-error-attributeerror-module-main-has-no-attribute) to reproduce this issue:

```python
from multiprocessing import Pool

def f(x):
    return x*x

if __name__ == '__main__':
    with Pool(5) as p:
        print(p.map(f, [1, 2, 3]))
```

And here is the full traceback:

```python
Traceback (most recent call last):

  File "<ipython-input-1-65a620cdba7c>", line 1, in <module>
    runfile('C:/Users/User/Desktop/test_multiprocess.py', wdir='C:/Users/User/Desktop')

  File "C:\Users\User\spyder-kernels\spyder_kernels\customize\spydercustomize.py", line 680, in runfile
    execfile(filename, namespace)

  File "C:\Users\User\spyder-kernels\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/User/Desktop/test_multiprocess.py", line 15, in <module>
    with Pool(5) as p:

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\context.py", line 119, in Pool
    context=self.get_context())

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\pool.py", line 174, in __init__
    self._repopulate_pool()

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\pool.py", line 239, in _repopulate_pool
    w.start()

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\process.py", line 105, in start
    self._popen = self._Popen(self)

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\context.py", line 322, in _Popen
    return Popen(process_obj)

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\popen_spawn_win32.py", line 33, in __init__
    prep_data = spawn.get_preparation_data(process_obj._name)

  File "C:\Anaconda3\envs\Spyder\lib\multiprocessing\spawn.py", line 172, in get_preparation_data
    main_mod_name = getattr(main_module.__spec__, "name", None)

AttributeError: module '__main__' has no attribute '__spec__'
```

**Comment by takumi-ricoh** [comment]
@jnsebgosselin Thanks for your comment. I hope this bug will be fixed.

**Comment by ccordoba12** [comment]
This will be fixed once we release our next version (3.3.2).

**Comment by chone666** [comment]
> This will be fixed once we release our next version (3.3.2).

Unfortunately, it wasn't.

**Comment by ccordoba12** [comment]
You also need to update spyder-kernels
